lib/src/migrators/module.dart

```diff
@@ -746,7 +745,11 @@ class _ModuleMigrationVisitor extends MigrationVisitor {
     var imports =
         partitionOnType<Import, StaticImport, DynamicImport>(node.imports);
     var staticImports = imports.item1;
-    var dynamicImports = imports.item2;
+    var dynamicImports = imports.item2.where((import) =>
+        !references.orphanImportOnlyFiles.contains(importCache
+            .canonicalize(Uri.parse(import.url),
+                baseImporter: importer, forImport: true)
+            ?.item2));
```

Comment:
Nit: I think this would be easier to read if you split it into separate statements.

```suggestion
    var dynamicImports = imports.item2.where((import) {
      var url = importCache.canonicalize(
              Uri.parse(import.url), baseImporter: importer, forImport: true)
          ?.item2;
      return !references.orphanImportOnlyFiles.contains(url);
    });
```